app/src/main/java/com/duckduckgo/app/browser/BrowserWebViewClient.kt

```diff
@@ -439,16 +439,21 @@ class BrowserWebViewClient @Inject constructor(
             }
         }
         val navigationList = webView.safeCopyBackForwardList() ?: return
-        val activeExperiments = contentScopeExperiments.getActiveExperiments()
-        webViewClientListener?.pageStarted(WebViewNavigationState(navigationList), activeExperiments)
+
+        appCoroutineScope.launch(dispatcherProvider.io()) {
```

Comment:
You can remove this launch and the changes within it (changing to main after), you are already doing that within the getActiveExperiments() method so you are launching twice.

Reply:
After discussing this on MM, I've done the following:
* Launch coroutine on `dispatcherProvider.main()`
* Switch to `dispatcherProvider.io()` within `getActiveExperiments()`
* Remove context switching back to `dispatcherProvider.main()` after getting active experiments

This way we reduce the number of context switches while still ensuring `getActiveExperiments()` doesn't block the UI thread regardless of which dispatcher it's called from